zaza/openstack/charm_tests/nova/tests.py

```diff
@@ -154,6 +154,21 @@ def test_930_check_virsh_default_network(self):
                 model_name=self.model_name)
             self.assertFalse(int(run['Code']) == 0)
 
+    def test_virsh_audit_action(self):
+        """Test virsh-audit action."""
+        for unit in zaza.model.get_units('nova-compute',
+                                         model_name=self.model_name):
+            logging.info('Running `virsh-audit` action'
+                         ' on  unit {}'.format(unit.entity_id))
+            action = zaza.model.run_action(
+                unit.entity_id,
+                'virsh-audit',
+                model_name=self.model_name,
+                action_params={})
+            if "failed" in action.data["status"]:
+                raise Exception(
+                    "The action failed: {}".format(action.data["message"]))
+
```

Comment:
If this is a new action on the charm, then this will break on older versions, meaning we can't land stable updates.  If this is the case, then this should be a new test class and that new class added to the revised version of the charm

Reply:
Hi Alex,  
Thanks for the comments. Yes, this is a new action,  I will add a new test class for it. And here is gerrit review link
https://review.opendev.org/c/openstack/charm-nova-compute/+/766412